relay-server/src/services/processor/playstation.rs

```diff
@@ -2,42 +2,328 @@
             && item.attachment_type() == Some(&AttachmentType::Prosperodump)
     }) {
-        // TODO: Add the expand logic here
+        let event = event.get_or_insert_with(Event::default);
+        let data = extract_data(&item.payload()).map_err(|err| {
+            ProcessingError::InvalidPlaystationDump(format!("Failed to extract data: {}", err))
+        })?;
+        let prospero_dump = ProsperoDump::parse(&data).map_err(|err| {
+            ProcessingError::InvalidPlaystationDump(format!("Failed to parse dump: {}", err))
+        })?;
+        let minidump_buffer = write_dump(&prospero_dump).map_err(|err| {
+            ProcessingError::InvalidPlaystationDump(format!("Failed to create minidump: {}", err))
+        })?;
+        update_sentry_event(event, &prospero_dump);
+
+        let mut item = Item::new(ItemType::Attachment);
+        item.set_filename("DO_NOT_USE");
+        item.set_payload(ContentType::Minidump, minidump_buffer);
+        item.set_attachment_type(AttachmentType::Minidump);
+        envelope.add_item(item);
+
+        for file in prospero_dump.files {
+            let mut item = Item::new(ItemType::Attachment);
+            item.set_filename(file.name);
+            item.set_attachment_type(AttachmentType::Attachment);
+
+            if let Some(content_type) = mime_guess::from_path(file.name)
+                .first()
+                .map(|m| ContentType::from(m.essence_str()))
+            {
+                item.set_payload(content_type, file.contents.to_owned());
+            } else {
+                item.set_payload_without_content_type(file.contents.to_owned());
+            }
+            envelope.add_item(item);
+        }
+
+        let mut console_log = prospero_dump.system_log.to_string();
+        for log_line in prospero_dump.log_lines {
+            console_log.push_str(log_line);
+        }
+        if !console_log.is_empty() {
+            let mut item = Item::new(ItemType::Attachment);
+            item.set_filename("console.log");
+            item.set_payload(ContentType::Text, console_log.into_bytes());
+            item.set_attachment_type(AttachmentType::Attachment);
+            envelope.add_item(item);
+        }
 
         metric!(counter(RelayCounters::PlaystationProcessing) += 1);
     }
 
-    Ok(())
+    Ok(Some(EventFullyNormalized(false)))
 }
 
-pub fn process(
-    _managed_envelope: &mut TypedEnvelope<ErrorGroup>,
-    _event: &mut Annotated<Event>,
-) -> Result<Option<EventFullyNormalized>, ProcessingError> {
-    // TODO: Add the processing logic here.
-    Ok(None)
+pub fn update_sentry_event(event: &mut Event, prospero: &ProsperoDump) {
+    let contexts = event.contexts.get_or_insert_with(Contexts::default);
+
+    let mut device_context = DeviceContext {
+        name: prospero
+            .system_name
+            .map(|s| Annotated::new(s.to_string()))
+            .unwrap_or_default(),
+        arch: Annotated::new("x86_64".into()),
```

Comment:
Similar here, use `to_owned`. `Into` makes sense when there is no more explicit conversion or if you are in a generic context.

Reply:
Makes sense